### PR TITLE
GIVCAMP-303 | Horizontal changemaker card; use modal for XS for all changemaker variants

### PR DESCRIPTION
**Pull request by yvonnetangsu**
# READY FOR REVIEW

# Summary
- Horizontal variant of changemaker card
- For all variants of changemaker cards, use modal for content for only the XS breakpoint.
- Add Headless UI package

# Review By (Date)
- Retro

# Criticality
- 6

# Review Tasks

## Setup tasks and/or behavior to test

1. Go to Storyblok Test -> Yvonne Test Horizontal Changemakers and select the PR286 URL from the visual editor
2. Look at both the horizontal and vertical changemaker cards at all breakpoints.
3. Test on the phone and make sure the modal works
4. Checks code for a11y
5. Test on Firefox (see call out below for the bug and the solution)

# Associated Issues and/or People
- JIRA ticket(s) - GIVCAMP-303

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,6 +32,7 @@
     "tsc": "tsc"
   },
   "dependencies": {
+    "@headlessui/react": "^2.0.3",
```
Comment (yvonnetangsu) [inline package.json:35]: Headless UI on version 2 now. Giving them another try. Looks like quite a lot of people are using it now 🤔

**Comment by sherakama** [comment]
Desktop:
Screenshot 2024-05-14 at 11 49 09

Mobile:
Screenshot_20240514-115004
share_767985792959039091

Not sure why Susan's card is larger than the others on the (md?) breakpoint. It looks a little odd.

**Comment by sherakama** [comment]
Can't reproduce the FF bug you described

**Comment by yvonnetangsu** [comment]
> Can't reproduce the FF bug you described

Oh did you try adding back container query locally to reproduce the bug? It's not in the preview anymore - got rid of it 😄

**Comment by sherakama** [comment]
Focus on the elements after interaction looks good
Lighthouse didn't report any errors
Keyboard nav is working well
Keyboard nav is contained to the modal when opened

Nice work.

**Comment by sherakama** [comment]
> > Can't reproduce the FF bug you described
> 
> Oh did you try adding back container query locally to reproduce the bug? It's not in the preview anymore - got rid of it 😄

I did not, and am fine with it as not reproducible.

File: package.json (hunk 1 above)
Comment (sherakama) [reply package.json:35]: Good to know. It was released not too long ago with a big update to their paid package as well.

**Review by sherakama — APPROVED** [review]

**Comment by yvonnetangsu** [comment]
> Not sure why Susan's card is larger than the others on the (md?) breakpoint. It looks a little odd.

Oh that's intentional @sherakama - The horizontal variant of the card has a 2x3 aspect ratio from SM to LG breakpoint, while the vertical ones stays fixed at 1x2. We designed the horizontal one to accommodate cards with longer body copy so it has to be wider from SM breakpoint. On XS everyone opens a modal so they can be the same size 😄

**Comment by yvonnetangsu** [comment]
@sherakama Putting the Netlify nextjs runtime plugin back in this PR. Pushing this to PROD next as Alexis said this is ready go to live 🤠

**Pull request merged**